### PR TITLE
[engraving] Moved draw... Step 37

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

File: src/engraving/rendering/dev/tdraw.cpp [hunk 1]
```diff
@@ -837,6 +845,36 @@ void TDraw::draw(const Expression* item, Painter* painter)
...
+                painter->setPen(Pen(FiguredBass::engravingConfiguration()->formattingMarksColor(), 3));
+                painter->drawLine(0.0, -2, len, -2);              // -2: 2 rast. un. above digits
+            }
+        }
+    }
+
+    if (item->items().size() < 1) {                                 // if not parseable into f.b. items
+        drawTextBase(item, painter);                                // draw as standard text
+    } else {
+        for (FiguredBassItem* item : item->items()) {               // if parseable into f.b. items
```
Comment (cbjeukendrup) [inline src/engraving/rendering/dev/tdraw.cpp:885]: There's a name clash here with `item`. Will lead to a compiler warning at least.

File: src/engraving/rendering/single/singledraw.cpp [hunk 2]
```diff
@@ -99,6 +101,9 @@ void SingleDraw::drawItem(const EngravingItem* item, draw::Painter* painter)
     case ElementType::EXPRESSION:   draw(item_cast<const Expression*>(item), painter);
         break;
 
+    case ElementType::FERMATA:      draw(item_cast<const Fermata*>(item), painter);
+        break;
+
```
Comment (cbjeukendrup) [inline src/engraving/rendering/single/singledraw.cpp:117]: FiguredBass will need to be handled here too: even though it is not in any of the palettes by default, the user can add it there by Ctrl+Shift+dragging it into a palette.

File: src/engraving/rendering/dev/tdraw.cpp (hunk 1 above)
Comment (igorkorsukov) [reply src/engraving/rendering/dev/tdraw.cpp]: fixed

File: src/engraving/libmscore/deadslapped.cpp
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply src/engraving/libmscore/deadslapped.cpp:1]: added

File: src/engraving/rendering/single/singledraw.cpp (hunk 2 above)
Comment (igorkorsukov) [reply src/engraving/rendering/single/singledraw.cpp:123]: thank you, added

**Review by cbjeukendrup — APPROVED** [review]

**Pull request merged**